src/Lib.P2P/PubSub/PubSubMessage.cs

```diff
@@ -0,0 +1,48 @@
+﻿using ProtoBuf;
+
+namespace Lib.P2P.PubSub
+{
+    /// <summary>
+    ///   The PubSub message exchanged between peers.
+    /// </summary>
+    /// <seealso ref="https://github.com/libp2p/specs/blob/master/pubsub/README.md"/>
+    [ProtoContract]
+    public class PubSubMessage
+    {
+        /// <summary>
+        ///   Sequence of topic subscriptions of the sender.
+        /// </summary>
+        [ProtoMember(1)]
+        public Subscription[] Subscriptions;
+
+        /// <summary>
+        ///   Sequence of topic messages.
+        /// </summary>
+        [ProtoMember(2)]
+        public PublishedMessage[] PublishedMessages;
+    }
+
+    /// <summary>
+    ///   A peer's subscription to a topic.
+    /// </summary>
+    /// <seealso ref="https://github.com/libp2p/specs/blob/master/pubsub/README.md"/>
+    [ProtoContract]
+    public class Subscription
+    {
+        /// <summary>
+        ///   Determines if the topic is subscribed to.
+        /// </summary>
+        /// <value>
+        ///   <b>true</b> if subscribing; otherwise, <b>false</b> if
+        ///   unsubscribing.
+        /// </value>
+        [ProtoMember(1)]
+        public bool Subscribe;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Make this field 'private' and encapsulate it in a 'public' property.](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4612924)